hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerReportHandler.java

```diff
@@ -187,7 +188,11 @@ public void onMessage(final ContainerReportFromDatanode reportFromDatanode,
             // This is a new Container not in the nodeManager -> dn map yet
             nodeManager.addContainer(datanodeDetails, cid);
           }
-          processSingleReplica(datanodeDetails, container, replica, publisher);
+          if (container == null || ContainerReportValidator
+                  .validate(container, replica)) {
+            processSingleReplica(datanodeDetails, container,
```

Comment:
As this is an exceptional case, we need to log something if the container fails validated, ideally with the source DN, container ID, the invalid replica index etc. Otherwise we are just dropping the containers silently, which could make things hard to debug.